### DISCUSSION TITLE
Vite Build: How to add <script> inside <div id="app"> </div>

### DISCUSSION THREAD

**Discussion by tohagan** [quasarframework/quasar#13702] [General - Components / Directives / etc] [answered]
In Webpack build I could use this code in `index.template.html`

```html

<body>
    <!-- DO NOT touch the following DIV -->
    <div id="q-app">
      <script>
        if (document.documentMode) {
          alert('WARNING: Internet Explorer is not supported. We recommend using either Microsoft Edge, Chrome or Firefox browsers or your Mobile Phone or an iPad/Tablet device.');
        }
      </script>
    </div>
  </body>
```

However in Vite builds the `index.html` no longer contains a `<div id="q-app">` element. So what alterative method can we use to add <script> code?

**Comment by yusufkandemir** [comment, marked as answer]
You don't need to put it inside that div. It also says `DO NOT touch` in the Webpack version. You can put it at the end of the body tag.

```html
<body>
  <!-- DO NOT touch the following DIV -->
  <div id="q-app"></div>

  <script>
    if (document.documentMode) {
      alert('WARNING: Internet Explorer is not supported. We recommend using either Microsoft Edge, Chrome or Firefox browsers or your Mobile Phone or an iPad/Tablet device.');
    }
  </script>
</body>
```

**Comment by alexrififi** [comment]
Why not put this code in `App.vue`?

**Reply by yusufkandemir** [reply]
Since IE is not supported, the app will not render at all due to errors. So, anything that runs inside boot files, `App.vue`, or any other Vue/JS/TS/whatever file will not work. So, the safest way of handling these unsupported browser alerts is to use an unprocessed/non-transpiled script with old code that is enough to be able to run on old browsers without errors, and it needs to run separately. There are better checks and ways of handling these instead of just showing an alert, but this is the idea behind it.

**Reply by tohagan** [reply]
Let me show you another example from another app ...  The objective is to show a message in case Vue fails to render or takes a long time to load.  So once Vue renders ... this div is replaced and that's the intention.

```html
<html>
  <head>
    ...
      <!-- Hides showWarning for 10 secs on load -->
      <style>
      #showWarning {
        opacity: 0;
        animation: cssAnimation 0s 10s forwards;
        -moz-animation: cssAnimation 10s;
        -webkit-animation: cssAnimation 10s;
        -o-animation: cssAnimation 10s;
        animation: cssAnimation 10s;
        -webkit-animation-fill-mode: forwards;
        animation-fill-mode: forwards;
      }
      @keyframes cssAnimation {
        99% { opacity: 0; }
        100% { opacity: 1; }
      }
      @-webkit-keyframes cssAnimation {
        99% { opacity: 0; }
        100% { opacity: 1; }
      }
    </style>
</head>
<body>
    <div id="q-app">
      <div id="showWarning">
        Sorry ... I'm not loading today 😔. Your network might be slow 🐌 or your <b>phone software</b> or <b>web browser</b> may need to be 👉 updated 👈.
      </div>
    </div>
</body>
</html>
```

**Comment by tohagan** [comment]
So for a viable Vite / Quasar solution ...  I suspect that I'd just use .. 

```
<body>
   <!-- quasar:entry-point -->

  <div id="showWarning">
    Sorry ... I'm not loading today 😔. Your network might be slow 🐌 or your <b>phone software</b> or <b>web browser</b> may need to be 👉 updated 👈.
  </div>
</body>
```

And then in App.vue we'd need include code to hide the `<div id="showWarning">` element once Vue loads. 

```
onMounted(() => {
  const el = document.getElementById('showWarning')
  el.parentNode.removeChild(el);
})

```
```
